projects/asn1-proto/asn1_proto_converter.cc

```diff
@@ -0,0 +1,134 @@
+#include "asn1_proto_converter.h"
+
+namespace asn1_proto {
+
+uint8_t ASN1ProtoConverter::GetNumBytes(size_t num) {
+  uint8_t shift = sizeof(size_t);
+  while (shift != 0) {
+    if (((num >> (shift * 8)) & 0xFF) != 0) {
+      return shift;
+    }
+    shift--;
+  }
+  return 0;
+}
+
+void ASN1ProtoConverter::AppendBytes(size_t num, size_t pos) {
+  uint8_t len_num_bytes = GetNumBytes(num);
+  std::vector<uint8_t> len_vec;
+  for (uint8_t i = len_num_bytes; i != 0; i--) {
+    len_vec.push_back((num >> (i * 7)) & 0xFF);
+  }
+  len_vec.push_back((num & 0xFF));
+  encoder_.insert(encoder_.begin() + pos, len_vec.begin(), len_vec.end());
+}
+
+size_t ASN1ProtoConverter::LongForm(const size_t assigned_len,
+                                    const size_t len_pos) {
+  uint8_t len_bytes = GetNumBytes(assigned_len);
+  if (assigned_len > 127) {
+    uint8_t longForm = (1 << 7);
+    longForm += len_bytes;
+    AppendBytes(longForm, len_pos);
```

Comment:
```suggestion
    uint8_t long_form = 0x80 | len_bytes
    AppendBytes(long_form, len_pos);
```

Or you could even collapse that further.

Reply:
I explicitly define long_form to make it clear what I'm passing in. If we can assume, that the reader should understand what we're doing here, then I can take out a lot of these middle steps.

Reply:
Yeah. I think this would be sufficient to add comments referencing the portion of the spec that this comes from, and a brief description.